tracing/src/main/java/com/palantir/tracing/NopDetachedSpan.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing;
+
+import com.palantir.tracing.api.SpanType;
+import java.util.Map;
+
+enum NopDetachedSpan implements DetachedSpan {
+    INSTANCE;
+
+    @Override
+    public CloseableSpan childSpan(String _operationName, SpanType _type) {
+        return NopCloseableSpan.INSTANCE;
```

Comment:
These methods should probably re-roll the dice in the current context rather than being no-ops.